### DISCUSSION TITLE
Changing screen using python/kivy

### DISCUSSION THREAD

**Discussion by magicduylove** [community/community#109304] [Programming Help]
### Select Topic Area

General

### Body

I just learned how to use kivy on Youtube and I came up with an idea of whenever I clicked on a button in the OneLineListItem it will move to that page, but I have tried many ways including screen_manager.current='page name'. But nothing seem work for me. Please, anyone show me what I can learn or something I should read to help me out this, I really stuck for a week with this. Thank you

```
`
my_screen='''
Screen:
    MDNavigationLayout:
        ScreenManager:
            Screen:
                BoxLayout:
                    orientation:'vertical'
                    MDTopAppBar:
                        title: 'Category'                
                        anchor_title: "center"
                        left_action_items:[["menu", lambda x: menuNav.set_state("open")]]
                    ScrollView:
                        MDList:
                            id:container
  
'''
```

class MainApp(MDApp):

     def build(self):
        self.screen=Builder.load_string(my_screen)           
        return self.screen

    def on_start(self):
        for i in self.caterogies:
            items=OneLineListItem(text=i)
            items.bind(on_release=self.change_screen_on_press)
            self.root.ids.container.add_widget(items)
        return
    
    def change_screen_on_press(self,instance):
        print(instance.text)
        self.screen=Builder.load_file('page.kv')
        return `

**Comment by alrnaldar** [comment]
It seems like your issue is that you're not switching to other screens when clicking on a list item. The problem might be related to the fact that you're loading a new KV file for each screen instead of using a screen manager to switch between them.

### Here's how you can fix your code:

1. Instead of loading a new KV file each time you want to switch to another screen, define all your screens inside one KV file, for example, app.kv.

2. Use a ScreenManager to manage transitions between screens.

3. When clicking on a list item, you can change the current screen using the switch_to method of your screen manager.

Here's an example of the corrected code:
```
from kivymd.app import MDApp
from kivy.lang import Builder

my_screen = '''
ScreenManager:
    id: screen_manager

    Screen:
        name: 'categories'
        BoxLayout:
            orientation: 'vertical'
            MDToolbar:
                title: 'Category'                
                anchor_title: "center"
                left_action_items: [["menu", lambda x: menuNav.set_state("open")]]
            ScrollView:
                MDList:
                    id: container
                    padding: dp(10)

    Screen:
        name: 'page'
        BoxLayout:
            orientation: 'vertical'
            MDToolbar:
                title: 'Page'
                anchor_title: "center"
                left_action_items: [["arrow-left", lambda x: app.go_back()]]
            MDLabel:
                text: 'This is the Page screen'
                halign: 'center'
                font_style: 'H4'
'''

class MainApp(MDApp):
    def build(self):
        return Builder.load_string(my_screen)

    def on_start(self):
        for category in self.categories:
            item = OneLineListItem(text=category)
            item.bind(on_release=self.change_screen_on_press)
            self.root.ids.container.add_widget(item)

    def change_screen_on_press(self, instance):
        print(instance.text)
        self.root.ids.screen_manager.current = 'page'

    def go_back(self):
        self.root.ids.screen_manager.current = 'categories'

if __name__ == '__main__':
    MainApp().run()
```
In this example, the ScreenManager contains two screens: "categories" and "page". When clicking on a list item in the "categories" screen, it switches to the "page" screen. When you're on the "page" screen, you can also go back to the "categories" screen by clicking the "arrow-left" button at the top of the screen.
